e2e/tests/transfer/upgradesv2_test.go

```diff
@@ -102,7 +102,7 @@ func (s *TransferChannelUpgradesTestSuite) TestChannelUpgrade_WithFeeMiddleware_
 		s.StartRelayer(relayer, testName)
 	})
 
-	s.Require().NoError(test.WaitForBlocks(ctx, 10, chainA, chainB), "failed to wait for blocks")
+	s.Require().NoError(test.WaitForBlocks(ctx, 20, chainA, chainB), "failed to wait for blocks")
```

Comment:
Num block waited was bumped here. Without this there was an error surfacing on L129 - when querying the channel version it was _not_ fee enabled. 

This must be some latency added somewhere in tx flow from the relayer trying to complete the channel upgrade. First it flushes packets and then makes this query against channel version to ensure the upgrade has completed.

Note, this could probably be changed to use `test.WaitForCondition` or something similar

Reply:
This failed again with the same on first CI run:

> failed to unmarshal metadata from version: ics20-2: invalid ICS29 middleware version

Best solution is to change to waitForCondition imo

Reply:
opened https://github.com/cosmos/ibc-go/issues/7691